collect_app/src/androidTest/java/org/odk/collect/android/storage/StorageMigratorTest.java

```diff
@@ -0,0 +1,115 @@
+package org.odk.collect.android.storage;
+
+import androidx.work.ExistingWorkPolicy;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import com.evernote.android.job.JobManager;
+import com.evernote.android.job.JobManagerCreateException;
+import com.evernote.android.job.JobRequest;
+
+import org.javarosa.core.reference.ReferenceManager;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.storage.migration.StorageEraser;
+import org.odk.collect.android.storage.migration.StorageMigrationRepository;
+import org.odk.collect.android.storage.migration.StorageMigrationResult;
+import org.odk.collect.android.storage.migration.StorageMigrator;
+import org.odk.collect.android.storage.utils.FakedAutoSendWorker;
+import org.odk.collect.android.storage.utils.FakedServerPollingJob;
+import org.odk.collect.android.tasks.ServerPollingJob;
+import org.odk.collect.android.upload.AutoSendWorker;
+
+import timber.log.Timber;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsNot.not;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+public class StorageMigratorTest {
```

Comment:
I'm wondering if there is a way we could test that the StorageMigrator does the right thing when these jobs are runnign without actually having to start them up (and also have tests that run in the JVM). As far as I can see these tests are non deterministic due to the waits and background work. 

I know Google released some testing around `WorkManager` so I can have a look and see if there is anything we can use there. Ideally we can fake the state of work or have it run paused or something. This wouldn't solve for the `JobManager` stuff though, Alternatively, for the moment we could hide the logic around looking up the statuses of these pieces of background work behind an interface that we can sub out for the `StorageMigrator`'s unit test (probably as a constructor arg):

```java
public interface WorkStatusProvider {
    public boolean isFormDownloaderRunning();
    public boolean isFormUploaderRunning();
}
```

This would also mean that objects like the `StorageMigrator` wouldn't have to care about the details involved of fetching those statuses.